### PR TITLE
GLTF module: Clean up lots of includes

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR cleans up includes in GLTF files. In a nutshell, here are the rules I'm working with for the includes:

* GLTFState includes all of the small classes that make it up.
  - Each of the small classes either has forward declarations or a full include of Godot classes. Which we use depends on if we need that class's full definition to make GLTFState and its small pieces work as-is.
* GLTFDocumentExtension includes GLTFState, since all document extensions need to use GLTFState.
* GLTFDocument includes GLTFDocumentExtension directly, which also includes GLTFState.
* The editor classes should only worry about GLTFDocument, the master class for running GLTF file imports.

So now there is a clear one-directional dependency: Editor -> GLTFDocument -> GLTFDocumentExtension -> GLTFState -> Small classes -> Godot classes. I said one-directional instead of linear because there are still some jumps, such as GLTFDocument needing GLTFState and the small classes. So, for example, none of the small classes are using GLTFState, and there is nothing in GLTFState that knows about GLTFDocument (except for `friend class`).

### CODE REVIEW

**Review by fire — APPROVED** [review]
Doesn't change behavior so it looks safe to merge.

**Comment by akien-mga** [comment]
Seems OK overall, but I'm unsure if all the changes go in the right direction. To improve code quality and clarify the include paths, we tend to go more in the direction of explicitly including the higher level includes that are actually needed by the class; e.g. if a class needs Node3D explicitly to do stuff with its API, it should include that header, and not only rely on the fact that it's included by a file that's included by a file that it includes.

We don't necessary go all the way down to like Variant or String or even `typedefs.h` with this logic, but for high level components it can make sense. There are no clear guidelines because it's not easy to put this into rules.

Feedback welcome from @Geometror @trollodel @KoBeWi.

**Comment by trollodel** [comment]
@akien-mga IMO, a good compromise is to accept implicit includes if:
- A required class is a superclass of another required class. Like Node, Node2D, and Node3D.
- The required class is part of the public (exposed) API of another required class. Like MeshInstance3D that includes Mesh.
- The required class is very unlikely to not be included anymore after a refactor. Like core/ classes.

File: modules/gltf/register_types.cpp [hunk 1]
```diff
@@ -30,23 +30,9 @@
 
 #include "register_types.h"
 
-#ifndef _3D_DISABLED
```
Comment (KoBeWi) [inline modules/gltf/register_types.cpp:33]: Why was this condition removed?

**Comment by KoBeWi** [comment]
> Otherwise if one of those dependency ends up no longer needed a given header, you can end up with compilation errors downstream because the class at the end of the chain wasn't actually including the header it explicitly needs.

From what I observed, dependency removals happen only while refactoring, at which point it's somewhat expected that things break and you have to do a bit more cleanup than was planned.

**Comment by Geometror** [comment]
I agree with @akien-mga, in general this looks like a good change, although I wouldn't replace the more explicit includes e.g. in gltf_document.cpp with the high-level ones. After all we want to reorganize the includes to reduce inclusion chains and speed up incremental builds.
Maybe a common header file which includes the most essential headers could mark a boundary: anything else needs to be explicitly included.

File: modules/gltf/register_types.cpp (hunk 1 above)
Comment (aaronfranke) [reply modules/gltf/register_types.cpp:33]: It did nothing before, the GLTF module already has checks in SCons to not compile when 3D is disabled.

**Comment by aaronfranke** [comment]
@Geometror If I'm understanding you correctly, then you would suggest I bring these three back? These are included farther down the chain which is why it currently compiles without them.

```cpp
#include "scene/animation/animation_player.h"
#include "scene/resources/importer_mesh.h"
#include "scene/resources/mesh.h"
```

**Comment by akien-mga** [comment]
I think we can merge as is as it's still a useful cleanup overall. The above discussion is more a start for coming up with clearer guidelines, but shouldn't block this specific cleanup.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**